test/test_hyper.py

```diff
@@ -275,6 +276,18 @@ def test_closed_connections_are_reset(self):
         assert c.recent_stream is None
         assert c.next_stream_id == 1
         assert c.window_manager is not wm
+        with c._conn as conn:
+            assert conn.state_machine.state == ConnectionState.IDLE
+
+        c = HTTP20Connection('www.google.com')
```

Comment:
@Lukasa I'm a little confused. The state of connection before or after `close()` should both be `IDLE`. But it holds the different `H2ConnectionStateMachine` instances.  So should I check the state of the origin state machine or manually change its state here?